app/src/main/java/ar/com/wolox/android/example/ui/home/HomePresenter.kt

```diff
@@ -4,4 +4,10 @@ import ar.com.wolox.android.example.utils.UserSession
 import ar.com.wolox.wolmo.core.presenter.BasePresenter
 import javax.inject.Inject
 
-class HomePresenter @Inject constructor(private val userSession: UserSession) : BasePresenter<HomeView>()
\ No newline at end of file
+class HomePresenter @Inject constructor(private val userSession: UserSession) : BasePresenter<HomeView>() {
+    override fun onViewAttached() {
+        if (!userSession.userIsLogged) {
+            view?.goToLoginPage()
```

Comment:
The best way to manage the session is to use a RootActivity, many times is also used as a SplashActivity, where you check if the user is logged or not. In that case, you go to the login or to home. If you check this in your Home view, you will always create the HomeActivity, HomeFragment and HomePresenter even before you know if you have to show the home or the login. Thats why it's a good practices to user a RootActivity that checks all of this.